design/node-storage-management.md

```diff
@@ -0,0 +1,46 @@
+# Support for Managing the Storage attached to the node.
+
+Node-bot is a infrastructure container (or a daemonset) that runs on each of the Kuberentes node. Node-bot is intended to be extensible and is to act as a bridge to augment the functionality already provided by the kubelet or other infrastructure containers running on the node. For example, node-bot will have to interface with node-problem-detector or heapster etc., 
+
+This is a design proposal is specificially for managing the storage attached to a Kubernetes node by a node-bot.  
+
+## Background
+
+Container Attached Storages like OpenEBS require a greater control in terms of accessing and managing the underlying storage infrastructure like SSDs, NVMe devices, etc., OpenEBS requires the following functionality to be available within a Kubernetes Cluster:
+- Need to know about the different types of disks available in the Cluster
+- Need to determine the physical location of the disks in the cluster in terms of Node, Rack, DataCenter/Zone
+- Need to associate the required storage to OpenEBS Storage Pods, taking into consideration the capacity, qos and availability requirements 
+- Need immediate notifications on the faults on the storage disks and ability to perform online corrections
```

Comment:
we also need to understand physical layout it terms of NUMA such that we can assign proper cores to the reactors, that's not on the top of the list right now though, but still we need something like this in the future

Reply:
Whats the format we are going to store in etcd?